### PR TITLE
remove delayed registering of IO events and IRQ FDs

### PR DESCRIPTION
**Pull request by acatangiu**
The `VmRequest` model is a leftover from crosvm's design where it is necessary because the devices and the vmm live in separate processes. This is not the case in Firecracker where we can directly register the device IO events and IRQ FDs.

Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: vmm/src/device_manager/mmio.rs [hunk 1]
```diff
@@ -108,18 +112,14 @@ impl MMIODeviceManager {
             let io_addr = IoeventAddress::Mmio(
                 self.mmio_base + u64::from(devices::virtio::NOTIFY_REG_OFFSET),
             );
-            self.vm_requests.push(VmRequest::RegisterIoevent(
-                queue_evt.try_clone().map_err(Error::CloneIoEventFd)?,
```
Comment (aghecenco) [inline vmm/src/device_manager/mmio.rs]: Looks like `CloneIoEventFd` and `CloneIrqFd` aren't used anywhere anymore, you can delete them.

Comment (acatangiu) [reply vmm/src/device_manager/mmio.rs]: @aghecenco good catch, done.

**Comment by dhrgit** [comment]
Will re-approve, after coverage threshold increase.

**Comment by andreeaflorescu** [comment]
FYI, the vsock timeout problem on `InstanceStart` seems to be back:
```API call exceeded maximum duration: 308.48 ms.```

**Pull request merged**